### PR TITLE
chore: cleanup overrides for the biomejs configuration

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

We discourage to override rules, if there are those should mention specific reasoning. 

**Description**

- Cleanup unnecessary override rules 
- Add explanation comments to all overriding rules
- Add proper naming conventions for all kinds of objects 

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

**Comment by nazarhussain** [comment]
Earlier most the conventions were disabled so we never check and validate. 

https://github.com/ChainSafe/lodestar/blob/c26cc63751f0a9f6f802153f002512dc5b8c58e5/.eslintrc.js#L54-L66

With biome, there is no way to disable formats, so why we have more detailed conventions list.

File: packages/api/src/utils/client/httpClient.ts [hunk 1]
```diff
@@ -196,7 +196,7 @@ export class HttpClient implements IHttpClient {
               this.logger?.debug("Requesting fallback URL", {routeId, baseUrl: printableUrl, score: this.urlsScore[i]});
             }
 
-            const i_ = i; // Keep local copy of i variable to index urlScore after requestWithBody() resolves
+            const localIndex = i; // Keep local copy of i variable to index urlScore after requestWithBody() resolves
```
Comment (nflaig) [inline packages/api/src/utils/client/httpClient.ts]: can't we just add a disable here? having the code reformatted into multiple lines is not great for readability here

**Review by nflaig — APPROVED** [review]
LGTM

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.23.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.23.0) :tada:

**Pull request merged**